clippy_lints/src/methods/mod.rs

```diff
@@ -4634,6 +4661,7 @@ impl Methods {
                         && matches!(kind, StmtKind::Semi(_))
                         && args.len() <= 1
                     {
+                        truncate_with_drain::check(cx, expr, recv, span, args.first());
```

Comment:
Shouldn't `clear_with_drain` get a chance to run first?

Reply:
I think the order is not important because `truncate_with_drain` don't lint on fully-opened range?

Reply:
I thought it would check a `0..` as well but it doesn't. Agreed, the order is unimportant in this case.